lib/sweet_xml.ex

```diff
@@ -143,6 +143,18 @@ defmodule SweetXml do
   @type spec :: %SweetXpath{}
   @type xmlElement :: record(:xmlElement)
 
+  defmodule XmerlFatal do
```

Comment:
Personal opinion, nested module makes the code hard to navigate, could both be extracted as `SweetXml.XmerlFatal` and `SweetXml.DTDError` ? I have barely played with exceptions in my career so I might say something stupid, but since a module name is an atom, it should work okay if there are multiple levels.

Reply:
Done :sunglasses: